build.sc

```diff
@@ -650,30 +636,43 @@ object dev extends MillModule{
     val isWin = scala.util.Properties.isWin
     val millPath = T.ctx.dest / (if (isWin) "mill.bat" else "mill")
     os.move(super.assembly().path, millPath)
-    if (isWin) windowsVmOptions("dev.launcher", millPath, forkArgs())
     PathRef(millPath)
   }
 
   def prependShellScript = T{
+    val (millArgs, otherArgs) = forkArgs().partition(arg => arg.startsWith("-DMILL") && !arg.startsWith("-DMILL_VERSION"))
+    // Pass Mill options via file, due to small max args limit in Windows
+    val vmOptionsFile = T.ctx.dest / "mill.properties"
+    val millOptionsContent = millArgs.map(_.drop(2).replace("\\", "/")).mkString("\r\n") // drop -D prefix, replace \ with /
+    os.write(vmOptionsFile, millOptionsContent)
+    val jvmArgs = otherArgs ++ List(s"-DMILL_OPTIONS_PATH=$vmOptionsFile")
     val classpath = runClasspath().map(_.path.toString)
-    val args = forkArgs()
-    val (shellArgs, cmdArgs) =
-      if (!scala.util.Properties.isWin) (
-        args,
-        args
-      )
-      else (
-        Seq("""-XX:VMOptionsFile="$( dirname "$0" )"/mill.vmoptions"""),
-        Seq("""-XX:VMOptionsFile=%~dp0\mill.vmoptions""")
-      )
     launcherScript(
-      shellArgs,
-      cmdArgs,
+      jvmArgs,
+      jvmArgs,
       classpath,
-      classpath
+      Agg(pathingJar().path.toString) // TODO not working yet on Windows! see #791
```

Comment:
On Windows, the classpath is only the "pathingJar", it contains all the dependencies.
It works in batch mode, interactive needs bit more debugging.

Reply:
@lihaoyi this only relates to the `dev.assembly` version..